word_segmentation.py

```diff
@@ -349,233 +413,225 @@ def preprocess_Burmese(demonstrate):
     """
-    This function first combine all lines in a file where each two lines are separated with a space, and then uses ICU
-    to segment the new long string.
-    Note: Because in some of the Burmese texts some lines start with glyphs that are not valid, I first combine all
-    lines and then segment them, rather than first segmenting each line and then combining them. This can result in a
-    more robust segmentation. Eample: see line 457457 of the my_train.txt
-    of the
+    This function uses a dataset with segmented lines to compute the accuracy of icu word breakIterator
     Args:
-        filename: address of the input file
+        filename: The path of the file
     """
-    words_break_iterator = BreakIterator.createWordInstance(Locale.getUS())
-    out_str = ""
     line_counter = 0
+    icu_mismatch = 0
+    icu_total_bies_lengths = 0
+    correctly_segmented_words = 0
+    true_words = 0
+    segmented_words = 0
     with open(filename) as f:
-        for line in f:
-            line_counter += 1
-            line = line.strip()
-            if is_ascii(line):
+        for file_line in f:
+            file_line = clean_line(file_line)
+            if file_line == -1:
                 continue
-            if len(out_str) == 0:
-                out_str = line
-            else:
-                out_str = out_str + " " + line
+            line = Line(file_line, "man_segmented")
+            true_bies = get_bies(line.char_brkpoints, line.man_word_brkpoints)
+            true_bies_str = get_bies_string_from_softmax(np.transpose(true_bies))
+            icu_bies = get_bies(line.char_brkpoints, line.icu_word_brkpoints)
+            icu_bies_str = get_bies_string_from_softmax(np.transpose(icu_bies))
 
-    words_break_iterator.setText(out_str)
-    icu_word_brkpoints = [0]
-    for brkpoint in words_break_iterator:
-        icu_word_brkpoints.append(brkpoint)
-    out_str = get_segmented_string(out_str, icu_word_brkpoints)
-    return out_str
+            # Counting the number of mismatches between icu_bies and true_bies
+            icu_total_bies_lengths += len(icu_bies_str)
+            icu_mismatch += diff_strings(true_bies_str, icu_bies_str)
+            line_counter += 1
+
+            # Computing the F1-score between icu_bies and true_bies
+            f1_output = compute_f1_score(true_bies=true_bies_str, est_bies=icu_bies_str)
+            correctly_segmented_words += f1_output[1]
+            true_words += f1_output[2]
+            segmented_words += f1_output[3]
 
+        icu_bies_accuracy = 1 - icu_mismatch / icu_total_bies_lengths
+        precision = correctly_segmented_words / segmented_words
+        recall = correctly_segmented_words / true_words
+        icu_f1_accuracy = 0
+        if precision + recall != 0:
+            icu_f1_accuracy = 2 * precision * recall / (precision + recall)
+        return icu_bies_accuracy, icu_f1_accuracy
 
-def get_file_text(filename):
+
+def normalize_bies(bies_str):
```

Comment:
If you had a first-class type for BIES, you could put functions like this one into that type.